tests/test-loaders.js

```diff
@@ -53,13 +54,31 @@ describe('test-loaders', function () {
         assert.isTrue(cssxref_found, data['macros']);
     })
 
+    it('The FileLoader should detect no macros', function () {
+        var macro_dir = 'tests/no-macros-in-here';
+        fs.mkdirSync(macro_dir);
+        try {
+            assert.throws(
+                function() {
+                    new ks_loaders.FileLoader({
+                        root_dir: macro_dir
+                    });
+                },
+                /no macros could be found in .+/
+            );
+        } finally {
+            fs.rmdirSync(macro_dir);
+        }
+    });
+
     it('The FileLoader should detect duplicate macros', function () {
         assert.throws(
             function() {
                 new ks_loaders.FileLoader({
                     root_dir: 'tests/fixtures'
                 });
-            }
+            },
+            /duplicate macros:[\s\S]+/
```

Comment:
Regex assertion of error messages is interesting. Python doesn't have it, I think. This is the first thing I've been jealous of.